library/pk.c

```diff
@@ -61,10 +61,11 @@ void mbedtls_pk_init( mbedtls_pk_context *ctx )
  */
 void mbedtls_pk_free( mbedtls_pk_context *ctx )
 {
-    if( ctx == NULL || ctx->pk_info == NULL )
+    if( ctx == NULL )
         return;
 
-    ctx->pk_info->ctx_free_func( ctx->pk_ctx );
+    if ( ctx->pk_info != NULL )
+        ctx->pk_info->ctx_free_func( ctx->pk_ctx );
```

Comment:
Doesn't this mean that if `ctx->pk_info` is `NULL`, then the `ctx` itself doesn't contain any sensitive information?
Do we really need to zeroize the ctx in this case?

Reply:
I think we tend to zeroize structures in `xxx_free()` functions regardless of whether their content is sensitive or not.

Reply:
Maybe. But that depends on the order of filling the structure, does it not?
It might be simpler and safer to clear any allocated memory than track execution paths when cleaning could be skipped.